packages/profile/src/components/trophies/pinned.tsx

```diff
@@ -0,0 +1,65 @@
+import {
+  cn,
+  SpiderWebIcon,
+  StateIconProps,
+  TrophyIcon,
+} from "@cartridge/ui-next";
+import { Card, CardHeader, CardTitle } from "@cartridge/ui-next";
+import { useMemo } from "react";
+
+export function Pinned({
+  Icon,
+  title,
+  empty,
+}: {
+  Icon: React.ComponentType<StateIconProps> | undefined;
+  title: string;
+  empty?: boolean;
+}) {
+  const AchievementIcon = useMemo(() => {
+    if (Icon) return Icon;
+    return TrophyIcon;
+  }, [Icon]);
+
+  return (
+    <Card>
+      <CardHeader
+        className={cn(
+          "flex flex-col justify-between items-center h-36 py-6",
+          empty && "bg-background border border-dashed border-secondary",
+        )}
+      >
+        <AchievementIcon
+          className={cn(
+            "min-w-12 min-h-12",
+            empty ? "opacity-10" : "text-primary",
+          )}
+          variant="solid"
+        />
+        <Title title={title} empty={empty} />
+      </CardHeader>
+    </Card>
+  );
+}
+
+function Title({ title, empty }: { title: string; empty?: boolean }) {
+  const overflow = useMemo(() => title.length > 27, [title]);
+  const content = useMemo(() => {
+    if (!overflow) return title;
+    return title.slice(0, 24) + "...";
```

Comment:
Can it be handled by `text-overflow` css?

https://tailwindcss.com/docs/text-overflow#truncate

Reply:
I was thinking it was not possible because I wanted the ellipsis on 2nd line, but actually I was able to make it works thanks to `line-clamp` that I didnt know, good catch

```js
break-words text-ellipsis line-clamp-2
```